### PR TITLE
Check dependencies for tutorials

### PR DESCRIPTION
**Pull request by jaakkonarhi**
# Description of the PR

+

# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [ ] new <b>unit</b> tests created</li>
        <li>- [ ] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [ ] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](https://github.com/Aalto-LeTech/aplus-courses/blob/master/TESTING.md) or other relevant documentation on _your_ branch?

- [ ] Yes.
- [ ] Not yet. I will do it next.
- [x] Not relevant.

### CODE REVIEW

File: src/main/java/fi/aalto/cs/apluscourses/intellij/notifications/MissingDependencyNotification.java [hunk 1]
```diff
@@ -6,18 +6,19 @@
 import com.intellij.notification.Notification;
 import com.intellij.notification.NotificationType;
 import fi.aalto.cs.apluscourses.intellij.services.PluginSettings;
+import java.util.Set;
 import org.jetbrains.annotations.NotNull;
 
 public class MissingDependencyNotification extends Notification {
   /**
    * Construct a missing module notification that explains that a module with the given name
```
Comment (Taikelenn) [inline src/main/java/fi/aalto/cs/apluscourses/intellij/notifications/MissingDependencyNotification.java:14]: Now that we have a set of module names, this should read:
"...notification which explains that modules with the given names couldn't be found."

File: src/main/java/fi/aalto/cs/apluscourses/intellij/actions/TutorialAction.java [hunk 2]
```diff
@@ -90,6 +96,21 @@ private void doTutorial(@NotNull Project project) {
     }
     TutorialExercise tutorialExercise = (TutorialExercise) selectedExercise.getModel();
 
+    Set<String> dependencies = tutorialExercise.getTutorial().getDependencies();
+    Set<String> missingDependencies = new java.util.HashSet<>();
```
Comment (Taikelenn) [inline src/main/java/fi/aalto/cs/apluscourses/intellij/actions/TutorialAction.java:100]: It feels weird to see a fully qualified default namespace `java.util` :p Perhaps add it to imports?

**Review by Taikelenn — APPROVED** [review]

File: src/main/java/fi/aalto/cs/apluscourses/intellij/actions/TutorialAction.java (hunk 2 above)
Comment (OlliKiljunen) [reply src/main/java/fi/aalto/cs/apluscourses/intellij/actions/TutorialAction.java:100]: I stand up for the "alternative" way Taikelenn described. Loops and mutable collections are something to be avoided whenever possible.

**Review by OlliKiljunen — APPROVED** [review]
👍

**Pull request merged**